provider/blockdev_commit_base.py

```diff
@@ -0,0 +1,153 @@
+                backup_utils.block_commit(self.main_vm, device, **arguments)
+            else:
+                commit_cmd = backup_utils.block_commit_qmp_cmd
+                cmd, args = commit_cmd(device, **arguments)
+                job_id = args.get("job-id", device)
+                job_id_list.append(job_id)
+                self.main_vm.monitor.cmd(cmd, args)
+        for job_id in job_id_list:
+            job_utils.wait_until_block_job_completed(self.main_vm, job_id)
+
+    @staticmethod
+    def get_linux_disk_path(session, disk_size):
+        disks = utils_disk.get_linux_disks(session, True)
+        for kname, attr in disks.items():
+            if attr[1] == disk_size and attr[2] == "disk":
+                return kname
+        return None
+
+    def configure_data_disk(self, tag):
+        os_type = self.params["os_type"]
+        disk_params = self.params.object_params(tag)
+        disk_size = disk_params["image_size"]
+        session = self.main_vm.wait_for_login()
+        try:
+            if os_type != "windows":
+                disk_id = self.get_linux_disk_path(session, disk_size)
+                assert disk_id, "Disk not found in guest!"
+                mount_point = utils_disk.configure_empty_linux_disk(
+                    session, disk_id, disk_size)[0]
+                self.disks_info.append([
+                    r"/dev/%s1" %
+                    disk_id, mount_point, tag])
+            else:
+                disk_id = utils_disk.get_windows_disks_index(
+                    session, disk_size)
+                driver_letter = utils_disk.configure_empty_windows_disk(
+                    session, disk_id, disk_size)[0]
+                mount_point = r"%s:\\" % driver_letter
+                self.disks_info.append([disk_id, mount_point, tag])
+        finally:
+            session.close()
+
+    def generate_tempfile(self, root_dir, filename="data",
+                          size="10M", timeout=360):
+        backup_utils.generate_tempfile(
+            self.main_vm, root_dir, filename, size, timeout)
+        self.files_info.append([root_dir, filename])
+
+    def pre_test(self):
+        if not self.main_vm.is_alive():
+            self.main_vm.create()
+        self.main_vm.verify_alive()
+        for device in self.params["device_tag"].split():
+            device_params = self.params.object_params(device)
+            snapshot_tags = device_params["snapshot_tags"].split()
+            self.device_node = self.get_node_name(device)
+            self.configure_data_disk(device)
+            self.prepare_snapshot_file(snapshot_tags)
+            self.create_snapshots(snapshot_tags, device)
+
+    def post_test(self):
+        try:
+            self.main_vm.destroy()
+            for image in self.snapshot_images:
+                image.remove()
+        except Exception as error:
+            logging.error(str(error))
+
+    def run_test(self):
+        self.pre_test()
```

Comment:
Please add 'kill_vm = yes' in cfg file in case some error might occur in pre_test

Reply:
OK, will update it.